### ISSUE TITLE
Javascript layer, evil been killed after switch from insert mode to normal mode

### ISSUE THREAD

**Issue by liujoey** [syl20bnr/spacemacs#1654]
When enabled the javascript layer on develop branch:

if I type some expression (with errors, e.g. no semi-colon after statement) and press ESC or fd to quite the insert mode, evil will be killed and there are some errors in the message buffer:

```
Error running timer `sp-show--pair-function': (error "Invalid search bound (wrong side of point)")
Error during redisplay: (eval (spacemacs/mode-line-prepare)) signaled (error "Invalid face" spacemacs-nil-face)
fringe-helper-modification-func: Invalid search bound (wrong side of point)Error during redisplay: (eval (spacemacs/mode-line-prepare)) signaled (error "Invalid face" spacemacs-nil-face)
Error during redisplay: (eval (spacemacs/mode-line-prepare)) signaled (error "Invalid face" spacemacs-nil-face)
missing ; after statement [4 times]
```

Every time this error occur, mode line will show nothing.
I have to M-x evil-normal-state to restart evil mode.

And backspace in insert mode is very slow, I need to hit backspace twice to delete one char.

Edit: problem been solved if I turn off the js2-mode errors and warnings by `SPC m w`, and the backspace issue was also been solved. Seems the js2-mode errors and warnings feature introduced a lot problems..

**Comment by liujoey** [comment]
After delete the  .cache folder, the problem solved. So I assume we need to remove cache folder every time after an update? 

For now I'll just close this issue

**Comment by liujoey** [comment]
No, sorry, it's not solved. After sometime, it appear again.

**Comment by ajnsit** [comment]
*bump*. I keep running into the same problem as well.

**Comment by TheBB** [comment]
Since we have stopped using git-gutter, I'm assuming the original problem is fixed. If not, please reopen or re-report. Same for your second issue, @ajnsit, since it's not what this thread is about.

**Comment by stormpat** [comment]
This issue seems to be back, on latest tagged release (v.0.105.8) 
It seems evil mode is killed, I cannot go back to normal mode, and i get the following error in the messages buffer:

```fringe-helper-modification-func: Invalid search bound (wrong side of point)```

**Comment by alxndr** [comment]
Saw this happening today with v0.105.11: The "Invalid search bound" message happened when I tried to save via `:w` (and it wouldn't save the file). I was not able to recall what I had done immediately beforehand, except that I have been messing around with an es6 linter.

I also noticed that when deleting several lines of arguments to a function and pasting them elsewhere, the yanked content seemed to be somewhat scrambled: if memory serves, it was the first and last lines which had been modified, using what seemed to be content from the rest of the yank (not verified, I can't recreate the issue now).
